### PR TITLE
Improve Image Gallery Upgrader

### PR DESCRIPTION
**Pull request by mamhoff**
It will now create a page-layout-specific "Gallery" element
that hold the individual images as nested elements. Image options
are parsed from the previous editor partials. The task is not idem-
potent as the views are rewritten after it runs.

### CODE REVIEW

File: lib/alchemy/upgrader/tasks/picture_gallery_upgrader.rb [hunk 1]
```diff
@@ -17,11 +17,27 @@ def convert_picture_galleries
...
+        config_picture_gallery_names_with_parent_names = elements_with_picture_gallery.map do |e|
+          [
+            e['name'],
+            "#{e['name']}_picture_gallery"
+          ]
+        end
+
+        all_picture_gallery_names_with_parent_names = config_picture_gallery_names_with_parent_names | db_picture_gallery_names_with_parent_names
+
+        if all_picture_gallery_names_with_parent_names.empty?
```
Comment (tvdeyen) [inline lib/alchemy/upgrader/tasks/picture_gallery_upgrader.rb]: You load to sets of elements union them but don't use them below this line. Why load them in the first place?

**Review by tvdeyen — COMMENTED** [review]
Can we remove the two sets of elements that are not used anymore?

**Comment by mamhoff** [comment]
Removed.

**Comment by tvdeyen** [comment]
Thank you so much

**Pull request merged**